examples/embeddings_extraction_s3e_pooling.py

```diff
@@ -0,0 +1,112 @@
+"""
+
+def fit(language_model, corpus_path, save_dir, do_lower_case, batch_size=4, use_gpu=False):
+    # Fit S3E on a corpus
+    set_all_seeds(seed=42)
+    device, n_gpu = initialize_device_settings(use_cuda=use_gpu, use_amp=False)
+
+    # Create a InferenceProcessor
+    tokenizer = Tokenizer.load(pretrained_model_name_or_path=language_model, do_lower_case=do_lower_case)
+    processor = InferenceProcessor(tokenizer=tokenizer, max_seq_len=128)
+
+    # Create an AdaptiveModel
+    language_model = LanguageModel.load(language_model)
+
+    model = AdaptiveModel(
+        language_model=language_model,
+        prediction_heads=[],
+        embeds_dropout_prob=0.1,
+        lm_output_types=["per_sequence"],
+        device=device)
+
+    model, processor, s3e_stats = fit_s3e_on_corpus(processor=processor,
+                                                    model=model,
+                                                    corpus=corpus_path,
+                                                    n_clusters=10,
+                                                    pca_n_components=300,
+                                                    svd_postprocessing=True,
+                                                    min_token_occurrences=1)
+
+    # save everything to allow inference without fitting everything again
+    model.save(save_dir)
+    processor.save(save_dir)
+    with open(save_dir / "s3e_stats.pkl", "wb") as f:
+        pickle.dump(s3e_stats, f)
+
+    # Load model, tokenizer and processor directly into Inferencer
+    inferencer = Inferencer(model=model, processor=processor, task_type="embeddings", gpu=use_gpu,
+                       batch_size=batch_size, extraction_strategy="s3e", extraction_layer=-1,
+                       s3e_stats=s3e_stats)
+
+    # Input
+    basic_texts = [
+        {"text": "a man is walking on the street."},
+        {"text": "a woman is walking on the street."},
+    ]
+
+    # Get embeddings for input text (you can vary the strategy and layer)
+    result = inferencer.inference_from_dicts(dicts=basic_texts)
+    print(result)
+
+
+def extract_embeddings(load_dir, use_gpu, batch_size):
+    with open(load_dir / "s3e_stats.pkl", "rb") as f:
+        s3e_stats = pickle.load(f)
+
+    # Init inferencer
+    inferencer = Inferencer.load(model_name_or_path=load_dir, task_type="embeddings", gpu=use_gpu,
+                       batch_size=batch_size, extraction_strategy="s3e", extraction_layer=-1,
+                       s3e_stats=s3e_stats)
+
+    # Input
+    basic_texts = [
+        {"text": "a man is walking on the street."},
+        {"text": "a woman is walking on the street."},
+    ]
+
+    # Get embeddings for input text
+    result = inferencer.inference_from_dicts(dicts=basic_texts)
+    print(result)
+
+
+if __name__ == "__main__":
+    #TODO update to public model / data
+    lang_model = Path("saved_models/s3e_fasttext")
```

Comment:
I know we said we want to have the path relative to top folder farm, but we never actually did. So lets use ../savedmodel and ../data, right?

Reply:
Fixed